docs/pages/machine-id/guides/github-actions.mdx

```diff
@@ -0,0 +1,173 @@
+with your own.
+
+Once the configuration YAML file has been created, apply it to your cluster using `tctl`:
+
+```code
+$ tctl create tokenconfig.yaml
+```
+
+The token `github-demo` can now be observed with the following command, ready to be used by
+Github:
+
+```code
+$ sudo tctl tokens ls
+```
+
+## Step 2/3. Creating your Machine ID bot
+
+With the join token for Github actions created, the next step is to create the Machine ID bot,
+and ensure that is configured to use the created token. This bot can be created using a Teleport
+`configuration.yaml` file, however for the purposes of this guide you will create a simple configuration
+using `tctl` with any required configuration flags.
+
+The Machine ID bot created in this example will be used to access a specific node on the cluster
+via `tsh ssh`, and will therefore require a role that can access the cluster as needed. This example
+configuration will apply the `access` role, however care should be taken to either create or apply
+a role according to the principle of least privelege in production environments. Additionally, it
+should have explicit access to the cluster using a username created specifically for the bot user alone.
+
+Enter the following command from your host with `tsh` access, replacing the `username` value with
+the username affiliated with a node the bot will be required to access over a `tsh ssh` session:
+
+```
+$ tctl bots add github-demo --roles=access --token=github-demo –logins=username
+```
+
+## Step 3/3. Configuring Github Actions
+
+Now that the bot has been successfully created, github actions can succesfully function as part of the Teleport cluster.
+
+In the Github workflows directory of your repo, generally `/.github/workflows/`, create a new workflow YAML file,
+in this case `actionstest.yml` that will reflect the action you'd like to configure. This guide will create an action that
+will both list nodes within the cluster using `tsh ls`, as well as write the commit SHA that triggered the workflow to
+a file within the bot node. To create this configuration, use the following YAML file:
+
+```YAML
+# This is a basic workflow to help you get started with Actionsname: Demo.
+# Will take the following action whenever a push is made to the "main" branch.
+on:
+ push:
+   branches:
+     - main
+jobs:
+ demo:
+   permissions:
+# This is required or tbot will not be able to authenticate.
+     id-token: write
+     contents: read
+# The name of the action, and the Linux distro to be used to perform the required steps.
+   name: guide-demo
+   runs-on: ubuntu-latest
+   steps:
+     - name: Checkout repository
+       uses: actions/checkout@v3
+     - name: Fetch Teleport binaries
+       uses: gravitational/teleport-actions/setup@main
+       with:
+         version: 11.0.0
+     - name: Fetch credentials
+       run: >
+         tbot start
+         --join-method=github
+         --token=github-demo
+# Use the address of the auth server for your own cluster.
```

Comment:
We should really lean towards telling people to use the address of their proxy here if possible ! This is more likely to work in more environments.